src/hpc/threadapi.c

```diff
@@ -432,15 +432,15 @@ static GVarDescriptor GVarTHREAD_EXIT;
 
 static void ThreadedInterpreter(void * funcargs)
 {
-    IntrState intr = { 0, 0, 0, 0 };
-    Obj tmp, func;
+    Obj tmp, func, oldLVars;
     int i;
 
-    /* initialize everything and begin an interpreter                       */
+    // initialize everything and begin a fresh execution context
     STATE(NrError) = 0;
     STATE(ThrownObject) = 0;
+    oldLVars = STATE(CurrLVars);
+    SWITCH_TO_OLD_LVARS(STATE(BottomLVars));
 
-    IntrBegin(&intr, STATE(BottomLVars));
```

Comment:
To understand why here and elsewhere, we can get rid of `IntrBegin` and `IntrEnd`, consider what its code:
```C
void IntrBegin(IntrState * intr, Obj frame)
{
    /* allocate a new values stack                                         */
    intr->StackObj = NEW_PLIST(T_PLIST, 64);

    /* must be in immediate (non-ignoring, non-coding) mode                */
    GAP_ASSERT(intr->ignoring == 0);
    GAP_ASSERT(intr->coding == 0);

    /* no return-statement was yet interpreted                             */
    intr->returning = 0;

    /* start an execution environment                                      */
#if BEFORE_THIS_PR
    ExecBegin(frame);
#else
    // remember the old execution state
    intr->oldLVars = STATE(CurrLVars);

    // start an execution environment
    SWITCH_TO_OLD_LVARS(frame);    
#endif
}
```

So all it really does is save the value of `STATE(CurrLVars);` and then invoke `SWITCH_TO_OLD_LVARS`; and as you see, we now exactly this here, instead of calling `IntrBegin`.

Similar for `IntrEnd`: All it does here, effectively (given that we even touch the `IntrState intr`, other than passing it to `IntrBegin` and `IntrEnd`, is to call `SWITCH_TO_OLD_LVARS`. Which we also do here (see new line 470 below).